neo/Ledger/MemoryPool.cs

```diff
@@ -286,32 +287,46 @@ internal bool TryAdd(UInt256 hash, Transaction tx)
 
             if (_unsortedTransactions.ContainsKey(hash)) return false;
 
+            List<Transaction> removedTransactions;
             _txRwLock.EnterWriteLock();
             try
             {
                 _unsortedTransactions.Add(hash, poolItem);
 
                 SortedSet<PoolItem> pool = tx.IsLowPriority ? _sortedLowPrioTransactions : _sortedHighPrioTransactions;
                 pool.Add(poolItem);
-                RemoveOverCapacity();
+                removedTransactions = RemoveOverCapacity();
             }
             finally
             {
                 _txRwLock.ExitWriteLock();
             }
 
+            foreach (IMemoryPoolTxObserverPlugin plugin in Plugin.TxObserverPlugins)
+            {
+                plugin.TransactionAdded(poolItem.Tx);
+                if (removedTransactions != null)
+                    plugin.TransactionsRemoved(MemoryPoolTxRemovalReason.CapacityExceeded, removedTransactions);
+            }
+
             return _unsortedTransactions.ContainsKey(hash);
         }
 
-        private void RemoveOverCapacity()
+        private List<Transaction> RemoveOverCapacity()
         {
+            List<Transaction> removedTransactions = null;
             while (Count > Capacity)
             {
                 PoolItem minItem = GetLowestFeeTransaction(out var unsortedPool, out var sortedPool);
 
                 unsortedPool.Remove(minItem.Tx.Hash);
                 sortedPool.Remove(minItem);
+                if (removedTransactions == null)
```

Comment:
I think I can get rid of having this check repeatedly in the while by only calling the method if it will remove transactions.